### PR TITLE
Add ScheduleFragment

### LINKED ISSUE #10: Events Activity
Should display the events in a ViewPager based on the day, and in a scrolling RecyclerView ordered by their time. Each event should have an icon that represents if notifications are turned on. Selecting the icon will result in a Snackbar that lets the user know they will receive push notifications about the event. Clicking on the event will open the event info page.

### PR DESCRIPTION
**Pull request by chyku**
Closes #10

### CODE REVIEW

File: app/src/main/java/org/hackillinois/android/view/schedule/ScheduleFragment.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    private ViewPager mViewPager;
+
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        ScheduleViewModel viewModel = ViewModelProviders.of(this).get(ScheduleViewModel.class);
+
+        viewModel.init();
+
+        viewModel.getEventsListLiveData().observe(this, new Observer<List<Event>>() {
```
[the remark this answers is not in the document]
Comment (chyku) [reply app/src/main/java/org/hackillinois/android/view/schedule/ScheduleFragment.java]: Thanks so much for the input- I'll get started on this asap!

File: app/src/main/java/org/hackillinois/android/view/schedule/EventsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        mEventList = eventsList;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent,
+                                         int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.schedule_event_list_item, parent, false);
+        ViewHolder viewHolder = new ViewHolder(view);
+        context = parent.getContext();
```
[the remark this answers is not in the document]
Comment (chyku) [reply app/src/main/java/org/hackillinois/android/view/schedule/EventsAdapter.java]: It's used for the favorite button when using FavoritesManager, in onBindViewHolder.

File: app/src/main/java/org/hackillinois/android/database/dao/EventDao.kt [hunk 3]
```diff
@@ -15,6 +15,9 @@ interface EventDao {
     @Query("SELECT * FROM events WHERE startTime <= :time AND endTime >= :time")
     fun getAllEventsHappeningAtTime(time: Long): LiveData<List<Event>>
 
+    @Query("SELECT * FROM events WHERE startTime >= :startTime AND startTime < :endTime")
```
Comment (patrickfeltes) [inline app/src/main/java/org/hackillinois/android/database/dao/EventDao.kt]: We can sort this in here by doing: `... startTime < :endTime ORDER BY :startTime`

File: app/src/main/java/org/hackillinois/android/view/schedule/ScheduleFragment.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            if (adapter != null) adapter.notifyDataSetChanged();
+        }
+
+        @Override
+        public void onCreate(Bundle savedInstanceState) {
+            super.onCreate(savedInstanceState);
+
+            int sectionNumber = getArguments() == null ? 0 : getArguments().getInt(ARG_SECTION_NUM);
+
+            ScheduleViewModel viewModel = ViewModelProviders.of(this).get(ScheduleViewModel.class);
```
Comment (patrickfeltes) [inline app/src/main/java/org/hackillinois/android/view/schedule/ScheduleFragment.java]: This is awesome! Thanks for making the change

File: app/src/main/java/org/hackillinois/android/database/entity/Event.kt [hunk 5]
```diff
@@ -13,7 +13,9 @@ data class Event(
     val locationDescription: String,
     val latitude: Double,
     val longitude: Double
-) : BaseEntity() {
+) : BaseEntity(), Comparable<Event> {
+
+    override fun compareTo(other: Event) = startTime.compareTo(other.startTime);
```
Comment (patrickfeltes) [inline app/src/main/java/org/hackillinois/android/database/entity/Event.kt]: This stuff can be removed once we sort in the SQL query

File: app/src/main/java/org/hackillinois/android/view/schedule/ScheduleFragment.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+        viewPager.setAdapter(sectionsPagerAdapter);
+        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
+        tabLayout.addOnTabSelectedListener(new TabLayout.ViewPagerOnTabSelectedListener(viewPager));
+
+        return view;
+    }
+
+
+    public static class DayFragment extends Fragment {
```
Comment (patrickfeltes) [inline app/src/main/java/org/hackillinois/android/view/schedule/ScheduleFragment.java]: Can this be moved into its own file in the schedule package?

**Comment by patrickfeltes** [comment]
The changes that you made look awesome! I added a few more comments of things that can be changed, but once those are done it should be ready!

**Pull request merged**